### ISSUE TITLE
Cleaned datasets uses too much memory

### ISSUE THREAD

**Issue by lisphilar** [lisphilar/covid19-sir#430]
## Summary
Cleaned datasets uses too much memory. This is due to the data types of `Country` and `Province` are `object`. They should be `category`. 

## (Optional) Related classes
`JHUData`, `PopulationData`, `OxCGRTData`, `LinelistData`, `CountryData`, `PCRData`

## Codes and outputs:
```Python
import covsirphy as cs
# Dataset preparation
data_loader = cs.DataLoader("input")
jhu_data = data_loader.jhu()
df = jhu_data.cleaned()
# Object types
df.info()
# Memory usage
df.memory_uage(deep=True)
```
This code returns 
```Python
<class 'pandas.core.frame.DataFrame'>
Int64Index: 361215 entries, 0 to 13509
Data columns (total 7 columns):
 #   Column     Non-Null Count   Dtype
---  ------     --------------   -----
 0   Date       361215 non-null  datetime64[ns]
 1   Country    361215 non-null  object
 2   Province   361215 non-null  object
 3   Confirmed  361215 non-null  int64
 4   Infected   361215 non-null  int64
 5   Fatal      361215 non-null  int64
 6   Recovered  361215 non-null  int64
dtypes: datetime64[ns](1), int64(4), object(2)
memory usage: 22.0+ MB
```
```Python
Index         2889720
Date          2889720
Country      23317366
Province     24580007
Confirmed     2889720
Infected      2889720
Fatal         2889720
Recovered     2889720
dtype: int64
```

## Environment
- CovsirPhy version: 2.13.1-alpha
- Python version; 3.8
- Installation: pipenv
- System: WSL (Ubuntu)

**Comment by lisphilar** [comment]
For `JHUData.cleaned()`, `df[["Country", "Province"]] = df[["Country", "Province"]].astype("category")` reduces 4 MB.

```Python
<class 'pandas.core.frame.DataFrame'>
Int64Index: 361215 entries, 0 to 13509
Data columns (total 7 columns):
 #   Column     Non-Null Count   Dtype
---  ------     --------------   -----
 0   Date       361215 non-null  datetime64[ns]
 1   Country    361215 non-null  category
 2   Province   361215 non-null  category
 3   Confirmed  361215 non-null  int64
 4   Infected   361215 non-null  int64
 5   Fatal      361215 non-null  int64
 6   Recovered  361215 non-null  int64
dtypes: category(2), datetime64[ns](1), int64(4)
memory usage: 18.0 MB
```
```Python
Index        2889720
Date         2889720
Country       745764
Province      823194
Confirmed    2889720
Infected     2889720
Fatal        2889720
Recovered    2889720
dtype: int64
```

**Comment by lisphilar** [comment]
This issue was solved with #431.
